### ISSUE TITLE
Multibindings broken with guice 5.1.0: BINDING_ALREADY_SET

### ISSUE THREAD

**Issue by josephlbarnett** [authzee/kotlin-guice#24]
The recently released guice 5.1.0 appears to change something that breaks the way the `KotlinMultibinder` binds the `Set` (and I think similar for the way the `KotlinMapBinder` binds the `Map`) and it appears that both the guice native multibinder and the kotlin-guice KotlinMultibinder are attempting to set the same binding.  Looking at the commit list, probably related to https://github.com/google/guice/commit/e6ec2a4a9433d64e763093a826eecf49f005aa5d and https://github.com/google/guice/commit/a260458b28b2dad9e2c8cfca2a2783e585784ab6 ?

```
  spek2:KotlinMultibinderSpec:KotlinMultibinder:@ProvidesIntoSet:binds complex types into an annotated set
    => com.google.inject.CreationException: Unable to create injector, see the following errors:
1) [Guice/BindingAlreadySet]: Set<? extends Callable<A>> annotated with @Annotated() was bound multiple times.
Bound at:
1  : KotlinMultibinderSpec$1$1$4$4$injector$1.provideAnnotatedACallable(KotlinMultibinderSpec.kt:298)
      \_ installed by: KotlinMultibinderSpec$1$1$4$4$injector$1 -> RealMultibinder
2  : KotlinMultibinderSpec$1$1$4$4$injector$1.provideAnnotatedACallable(KotlinMultibinderSpec.kt:298)
      \_ installed by: KotlinMultibinderSpec$1$1$4$4$injector$1 -> RealKotlinMultibinder
Learn more:
  https://github.com/google/guice/wiki/BINDING_ALREADY_SET
2) [Guice/BindingAlreadySet]: Set<? extends Callable<A>> was bound multiple times.
Bound at:
1  : KotlinMultibinderSpec$1$1$4$4$injector$1.provideUnannotatedACallable(KotlinMultibinderSpec.kt:309)
      \_ installed by: KotlinMultibinderSpec$1$1$4$4$injector$1 -> RealMultibinder
2  : KotlinMultibinderSpec$1$1$4$4$injector$1.provideUnannotatedACallable(KotlinMultibinderSpec.kt:309)
      \_ installed by: KotlinMultibinderSpec$1$1$4$4$injector$1 -> RealKotlinMultibinder
Learn more:
  https://github.com/google/guice/wiki/BINDING_ALREADY_SET
2 errors
======================
Full classname legend:
======================
A:                                        "dev.misfitlabs.kotlinguice4.multibindings.A"
Annotated:                                "dev.misfitlabs.kotlinguice4.multibindings.Annotated"
Callable:                                 "java.util.concurrent.Callable"
KotlinMultibinderSpec$1$1$4$4$injector$1: "dev.misfitlabs.kotlinguice4.multibindings.KotlinMultibinderSpec$1$1$4$4$injector$1"
RealKotlinMultibinder:                    "dev.misfitlabs.kotlinguice4.multibindings.RealKotlinMultibinder"
RealMultibinder:                          "com.google.inject.internal.RealMultibinder"
========================
End of classname legend:
========================
       com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:576)
       com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:163)
       com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:110)
       com.google.inject.Guice.createInjector(Guice.java:87)
       com.google.inject.Guice.createInjector(Guice.java:69)
       com.google.inject.Guice.createInjector(Guice.java:59)
       dev.misfitlabs.kotlinguice4.multibindings.KotlinMultibinderSpec$1$1$4$4.invoke(KotlinMultibinderSpec.kt:290)
       dev.misfitlabs.kotlinguice4.multibindings.KotlinMultibinderSpec$1$1$4$4.invoke(KotlinMultibinderSpec.kt:37)
       org.spekframework.spek2.runtime.scope.TestScopeImpl.execute(Scopes.kt:94)
       org.spekframework.spek2.runtime.Executor$execute$$inlined$executeSafely$lambda$1$1.invokeSuspend(Executor.kt:52)
       [...]
```

**Comment by suman-ganta** [comment]
I pulled `dev.misfitlabs.kotlinguice4:kotlin-guice:1.5.0` but that did not include the fix for this one. Am I missing something?

**Comment by josephlbarnett** [comment]
1.5.0 predates this fix, we need a new version with this change to be released.

**Comment by suman-ganta** [comment]
Thanks @josephlbarnett. Any ETA for the next release?

**Comment by johnlcox** [comment]
I'm hoping to get this released this weekend and to include #26 in that release as well.

**Comment by johnlcox** [comment]
1.6.0 has been released and pushed to maven central.
